lib/srv/termmanager.go

```diff
@@ -108,12 +109,20 @@ func (g *TermManager) writeToClients(p []byte) {
 
 			// Let term manager decide how to handle broken party writers
 			if g.OnWriteError != nil {
+				// writeToClients is called with the lock held, so we need to release it
+				// before calling OnWriteError to avoid a deadlock if OnWriteError
+				// calls DeleteWriter/DeleteReader.
+				g.mu.Unlock()
 				g.OnWriteError(key, err)
+				g.mu.Lock()
 			}
 
-			delete(g.writers, key)
+			toDelete = append(toDelete, key)
```

Comment:
Are there any consequences of delaying this delete to the end?

Reply:
I reverted it. It was a left over from something I was doing to figure out the problem.